components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -458,7 +460,9 @@ class GeckoEngineSession(
                 interceptor != null && (!request.isDirectNavigation || interceptor.interceptsAppInitiatedRequests())
             ) {
                 val engineSession = this@GeckoEngineSession
-                val isSameDomain = engineSession.currentUrl?.tryGetHostFromUrl() == request.uri.tryGetHostFromUrl()
+                val isSameDomain = lastOnLoadRequestUrl?.tryGetHostFromUrl()?.toDisplayUrl() ==
+                    request.uri.tryGetHostFromUrl().toDisplayUrl()
+                lastOnLoadRequestUrl = request.uri
```

Comment:
Why do we need to introduce new state here? I assume `currentUrl` is not set yet because `onLocationChange` wasn't called yet? Any chance we can prevent adding this? 

Reply:
My concern with this is that the last `loadRequest` could have failed and then our check wouldn't be accurate anymore. Right now we compare the `currentUrl` with the requested on which is accurate.

Reply:
We talked it through and this is in fact the problem.  Let's double check with GV that there's now way for us to get to the `triggerUri` in this case or maybe file a ticket?